### PR TITLE
add GitHub Templates for issues

### PR DESCRIPTION
**Pull request by jennyf19**
#1486 

I look the same format we have in Id.Web and MSAL.NET.

[ASP.NET Core](https://github.com/dotnet/aspnetcore/tree/main/.github/ISSUE_TEMPLATE) as a few more templates, like for API/design proposals. If you want something similar, lmk.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+assignees: ''
+
+---
+
+**Which version of Microsoft.IdentityModel are you using?**
+Note that to get help, you need to run the latest version. 
+<!-- E.g. Microsoft.IdentityModel 6.14 -->
+
+**Where is the issue?**
+ * [ ] M.IM.JsonWebTokens
```
Comment (jennyf19) [inline .github/ISSUE_TEMPLATE/bug_report.md:15]: I'm not familiar with the type of issues opened in Wilson, so maybe you want something different listed here? in [id. web](https://github.com/AzureAD/microsoft-identity-web/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) we have them listed by scenario.

@brentschmaltz @mafurman

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * [ ] M.IM.Protocols.OpenIdConnect
+ * [ ] M.IM.Tokens
+ * [ ] M.IM.Tokens.Jwt
+ * [ ] M.IM.Validators
+* Other (please describe)
```
Comment (TimHannMSFT) [inline .github/ISSUE_TEMPLATE/bug_report.md]: nit: spacing on '*' looks a bit off, perhaps continue the [ ] syntax here too for consistency?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+A clear and concise description of what you expected to happen (or code).
+
+**Actual behavior**
+A clear and concise description of what happens, e.g. an exception is thrown, UI freezes.
+
+**Possible solution**
+<!--- Only if you have suggestions on a fix for the bug. -->
+
+**Additional context / logs / screenshots / links to code**
+Add any other context about the problem here, such as logs and screenshots or links to code.
```
Comment (TimHannMSFT) [inline .github/ISSUE_TEMPLATE/bug_report.md:55]: nit: Might be worth reminding folks not to include an PII (as we do in ICM templates etc)

**Review by TimHannMSFT — APPROVED** [review]
LGTM, not familiar with these sorts of templates though so take that with a grain of salt ;)

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (jennyf19) [reply .github/ISSUE_TEMPLATE/bug_report.md]: might be a good idea, yes. lol.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (jennyf19) [reply .github/ISSUE_TEMPLATE/bug_report.md:55]: good point. will add in the other repos as well.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+A clear and concise description of what you expected to happen (or code).
+
+**Actual behavior**
+A clear and concise description of what happens, e.g. an exception is thrown, UI freezes.
+
+**Possible solution**
+<!-- Only if you have suggestions on a fix for the bug. -->
+
+**Additional context / logs / screenshots / links to code**
+<!-- Please do not include any customer data or Personal Identifiable Information (PII) in any content posted to GitHub. See https://docs.microsoft.com/en-us/compliance/regulatory/gdpr#gdpr-faqs for more info on PII.-->
```
Comment (jmprieur) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
<!-- Please do not include any customer data or Personal Identifiable Information (PII) in any content posted to GitHub. See https://docs.microsoft.com/compliance/regulatory/gdpr#gdpr-faqs for more info on PII.-->
```

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @jennyf19

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (brentschmaltz) [reply .github/ISSUE_TEMPLATE/bug_report.md]: might want to add a check for each of our assemblies.
M.IM.Logging, M.IM.Protocols, M.IM.P.OpenIdConnect, M.IM.P.SignedHttpRequest, M.IM.P.WsFederation, M.IM.Tokens.Saml, M.IM.Xml

Note it is S.IM.Tokens.Jwt not M.IM.Tokens.Jwt

Comment (jennyf19) [reply .github/ISSUE_TEMPLATE/bug_report.md]: fixed.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -13,10 +13,19 @@ Note that to get help, you need to run the latest version.
 
 **Where is the issue?**
  * [ ] M.IM.JsonWebTokens
+ * [ ] M.IM.Protocols
  * [ ] M.IM.Protocols.OpenIdConnect
+ * [ ] M.IM.Protocols.SignedHttpRequest
+ * [ ] M.IM.Protocols.WsFederation
  * [ ] M.IM.Tokens
- * [ ] M.IM.Tokens.Jwt
+ * [ ] S.IM.Tokens.Jwt
```
Comment (brentschmaltz) [inline .github/ISSUE_TEMPLATE/bug_report.md:21]: Could we put the items in lexicographical order?

Comment (jennyf19) [reply .github/ISSUE_TEMPLATE/bug_report.md:21]: yes. will fix!

Comment (jennyf19) [reply .github/ISSUE_TEMPLATE/bug_report.md:21]: #1760

**Pull request merged**